app/routes.py

```diff
@@ -1,2 +1,58 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request, abort
+from app.routes_helpers import validate_model
 
+planets_bp = Blueprint('planets', __name__, url_prefix='/planets')
+
+@planets_bp.route("", methods=["GET"])
+def read_all_planets():
+    name_query = request.args.get("name")
+
+    if name_query:
+        planets = Planet.query.filter_by(name=name_query)
+    else:
+        planets = Planet.query.all()
+
+    planets_response = [planet.to_dict() for planet in planets]
+
+    return jsonify(planets_response)
+
+@planets_bp.route("", methods=["POST"])
+def create_planet():
+    request_body = request.get_json()
+    new_planet = Planet.from_dict(request_body)
+
+    db.session.add(new_planet)
+    db.session.commit()
+
+    return make_response(f"Planet {new_planet.name} successfully created", 201)
+
+@planets_bp.route("/<planet_id>", methods=["GET"])
+def read_one_planet(planet_id):
+    planet = validate_model(Planet, planet_id)
+
+    return planet.to_dict()
+
+@planets_bp.route("/<planet_id>", methods=["PUT"])
+def update_planet(planet_id):
+    planet = validate_model(Planet, planet_id)
+
+    request_body = request.get_json()
+
+    planet.name = request_body["name"]
+    planet.description = request_body["description"]
+    planet.distance_from_the_sun = request_body["distance_from_the_sun"]
+
+    db.session.commit()
+
+    return make_response(f"Planet #{planet.id} successfully updated")
```

Comment:
👀 At the least, this should `jsonify` the result so that all our endpoints return JSON. However, as with the POST endpoint, we might want to return the updated record.